src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp

```diff
@@ -1072,33 +1077,39 @@ void DiskObjectStorageTransaction::commit(const TransactionCommitOptionsVariant
                 getLogger("DiskObjectStorageTransaction"),
                 fmt::format("An error occurred while executing transaction's operation #{} ({})", i, operations_to_execute[i]->getInfoForLog()));
 
-            for (int64_t j = i; j >= 0; --j)
+            try
             {
-                try
-                {
-                    operations_to_execute[j]->undo();
-                }
-                catch (...)
-                {
-                    tryLogCurrentException(
-                        getLogger("DiskObjectStorageTransaction"),
-                        fmt::format("An error occurred while undoing transaction's operation #{}", i));
-
-                    throw;
-                }
+                undo();
```

Comment:
you can't call undo here, because it will rollback all the operations instead of partially executed prefix.